src/components/atoms/button.js

```diff
@@ -0,0 +1,45 @@
+import React, { Children } from "react"
+
+const Button = React.forwardRef(
+  (
+    {
+      variant = "primary",
+      size = "large",
+      type = "button",
+      onClick = null,
+      disabled = false,
+      className = "",
+      loading = false,
+      children,
+      ...props
+    },
+    ref
+  ) => {
+    const handleClick = e => {
+      if (!loading && onClick) {
+        onClick(e)
+      }
+    }
+
+    return (
+      <button
+        className={`btn-${variant}-${size} ${className}`}
+        onClick={handleClick}
+        disabled={disabled || loading}
+        type={type}
+        ref={ref}
+        {...props}
+      >
+        {Children.map(children || null, (child, i) => {
```

Comment:
nit: I am quite sure, that `Children.map` will handle `children` being `null` gracefully